src/components/Trade/hooks/useSwapper/getTradeQuoteArgs.tsx

```diff
@@ -0,0 +1,70 @@
+import type { Asset } from '@shapeshiftoss/asset-service'
+import type { UtxoBaseAdapter } from '@shapeshiftoss/chain-adapters'
+import type { HDWallet } from '@shapeshiftoss/hdwallet-core'
+import type { UtxoSupportedChainIds } from '@shapeshiftoss/swapper'
+import type { BIP44Params, UtxoAccountType } from '@shapeshiftoss/types'
+import {
+  isSupportedNonUtxoSwappingChain,
+  isSupportedUtxoSwappingChain,
+} from 'components/Trade/hooks/useSwapper/typeGuards'
+import type { TradeQuoteInputCommonArgs, TS } from 'components/Trade/types'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { toBaseUnit } from 'lib/math'
+
+export type GetTradeQuoteInputArgs = {
+  sellAsset: Asset
+  buyAsset: Asset
+  sellAccountType: UtxoAccountType | undefined
+  sellAccountBip44Params: BIP44Params
+  wallet: HDWallet
+  receiveAddress: NonNullable<TS['receiveAddress']>
+  sellAmountBeforeFeesCryptoPrecision: string
+  isSendMax: boolean
+}
+
+export const getTradeQuoteArgs = async ({
```

Comment:
Existing logic, extracted into a pure function to consolidate commonly used logic and prevent circular imports.